src/lib/deeplink/handlers/add-site-blueprint-with-url.ts

```diff
@@ -0,0 +1,66 @@
+import { app, dialog } from 'electron';
+import nodePath from 'path';
+import { __ } from '@wordpress/i18n';
+import fs from 'fs-extra';
+import { sendIpcEventToRenderer } from 'src/ipc-utils';
+import { download } from 'src/lib/download';
+import { getMainWindow } from 'src/main-window';
+
+/**
+ * Handles the add-site deeplink callback.
+ * This function is called when a user clicks a deeplink like:
+ * wpcom-local-dev://add-site?blueprint_url=<encoded-url>
+ *
+ * It downloads the blueprint from the URL, saves it locally, and opens the Add Site modal
+ * with the blueprint pre-filled.
+ */
+export async function handleAddSiteBlueprintWithUrl( urlObject: URL ): Promise< void > {
```

Comment:
I think we could consider making the `add-site` deeplink handler name more universal, i.e. that it could semantically cover both cases:
- blueprint from linked JSON URL
- blueprint that's a part of URL

And also keep the `Deeplink` in the name.

For example, it could be called `handleAddSiteDeeplink` and then we would just distinguish two parameters in it (`blueprint_url` and `blueprint`).